src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyFacet.java

```diff
@@ -139,8 +147,8 @@ private Content putMetadata(final String assetPath,
       TempBlob blob = tempBlob;
       switch (assetKind) {
         case DOWNLOAD_URL:
-          blob = absoluteUrlRemover.removeAbsoluteUrls(tempBlob, getRepository());
-          // break intentionally left out!
+          String assetName = group + "/" + project + "/" + version + "/index.json";
+          blob = absoluteUrlRemover.updateAbsoluteUrls(tempBlob, getRepository(), assetName);
```

Comment:
There is a potential security issue here if we aren't validating the hash of the download_url file (are we? I saw hashVerifier but that looked like it was verifying much later). Basically if NXRM was hit by a Man-in-the-middle-attack then it could end up using a URL that isn't the remote url configured against the repo which would then allow an attacker to return packages from their own server.

Reply:
Unfortunately conan does not supply a hash of the download_ulrl anywhere AFAICT so it is an issue with conan itself

Reply:
Does the download_url (or any other file) provide hashes of the packages?

Reply:
The conanmanifest file contains the hash of the conanfile.py and (I believe ) itself.